cpp/include/cudf/detail/rolling.hpp

```diff
@@ -48,5 +48,14 @@ std::unique_ptr<column> rolling_window(column_view const& input,
                                        rmm::cuda_stream_view stream,
                                        rmm::device_async_resource_ref mr);
 
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> windows_from_offset(
+  column_view const& input,
+  scalar const& length,
+  scalar const& offset,
+  window_type const window_type,
+  bool only_preceding,
```

Comment:
Can this also be a `const` ?

Reply:
I don't think we should make copy-by-value parameters like this `const`. 
This is not needed by the caller and provides too much limitation to the internal implementation.

Reply:
Point taken, @davidwendt.  For my own understanding, should the preceding parameter (`window_type`) be subject to the same convention?

Maybe it doesn't apply, since it isn't obviously an enum?

Reply:
IMO, I think that it should apply to this enum parameter as well.